### PR TITLE
Add minimal GHA pipeline

### PR DESCRIPTION
**Pull request by ssbarnea**
Most changes are inspired from vscode-ansible extension as we tried
to minimize divergence between their configs.

### CODE REVIEW

**Review by webknjaz — CHANGES_REQUESTED** [review]
blocking this for now because I have something locally and it should be built on top of that.

File: package.json
Comment (webknjaz) [inline package.json:27]: v14 is still active and even v12 is an LTS. So I wouldn't set this too high. Also, this is clearly a separate change so I expect it to be done in a separate PR.

File: package.json
Comment (webknjaz) [inline package.json:74]: Same here, removing yarn has nothing to do with GHA.

File: package-lock.json
Comment (webknjaz) [inline package-lock.json:2]: This is addressed in #11, it should be separate.

File: README.md
Comment (webknjaz) [inline README.md:15]: This needs to be in a PR removing yarn.

File: .npmrc
Comment (webknjaz) [inline .npmrc:1]: This should be explained and added separately.

File: .github/workflows/ci.yaml
Comment (webknjaz) [inline .github/workflows/ci.yaml:1]: I expect this to be moved to the workflow added in #18 once that is merged.

File: .github/workflows/ci.yaml
Comment (webknjaz) [inline .github/workflows/ci.yaml:11]: FWIW I don't think we need to filter branches.

File: .github/workflows/ci.yaml
Comment (webknjaz) [inline .github/workflows/ci.yaml:35]: I prefer seeing a distinct log entry for this. Also, since it's not used anywhere, no need to integrate it in GHA prematurely.

File: .github/workflows/ci.yaml
Comment (webknjaz) [inline .github/workflows/ci.yaml:42]: I think this will need to go to a "devenv" job.

File: .github/workflows/ci.yaml
Comment (webknjaz) [inline .github/workflows/ci.yaml:54]:
```suggestion
      - name: Ensure the job does not produce non-checked-in files
```

**Review by webknjaz — CHANGES_REQUESTED** [review]
Here's a short feedback with places that need to be atomic or changed. But I suggest discussing the GHA specifics after #19 and #20 are in. Other things seem to be mergeable mostly w/o debates.

File: .gitattributes
Comment (webknjaz) [inline .gitattributes:9]: Oh, and this one is separate too.

File: .github/workflows/ci.yaml
Comment (ssbarnea) [reply .github/workflows/ci.yaml:11]: In fact it is needed because otherwise it will duplicate the jobs on any development branch, causing noise and making harder to read the checks.

If we will ever have a `stable/` branch we can add it to the list but I doubt it will be too soon, if ever.

Another option to avoid duplicate job runs is to remove the push condition entirely. Still that is not ideal as we may want to enable a scheduler for building main every day or week, so we would know the status of the master branch, which could deprecate for external reasons.

I checked other projects where I remember having the duplicate problem in the past and they seem to have kinda similar setup. https://github.com/ansible-community/molecule/blob/main/.github/workflows/tox.yml#L7-L10 or https://github.com/ansible-community/ansible-lint/blob/main/.github/workflows/tox.yml#L7-L10

Whatever we pick, we should ensure that we avoid duplicate check runs on PRs, regardless if it comes from a fork or from inside the repository.

File: .github/workflows/ci.yaml
Comment (ssbarnea) [reply .github/workflows/ci.yaml]: Removed

File: .github/workflows/ci.yaml
Comment (ssbarnea) [reply .github/workflows/ci.yaml]: done

File: package-lock.json
Comment (ssbarnea) [reply package-lock.json:2]: I took it out, is now part of #25

File: .github/workflows/ci.yaml
Comment (webknjaz) [reply .github/workflows/ci.yaml:9]: As explained in https://github.com/ansible/ansible-language-server/pull/18#discussion_r709876416 and https://github.com/ansible/ansible-language-server/pull/18/files#r710255129 — the filtering is actually harmful to our developer experience for as long as we keep using upstream for our feature branches. But I'll reiterate here — it's should be a dedicated discussion separate from this PR. I firmly believe that we mustn't filter out events until we make the DX suitable for such a setup.
So while I recognize the problem you're attempting to address, I don't believe that we should rush into replacing one annoyance with the other.

File: .github/workflows/ci.yaml
Comment (webknjaz) [inline .github/workflows/ci.yaml:13]: I don't think we need an annotated config. Comments should only explain motivation and weird corner cases.

File: .github/workflows/ci.yaml
Comment (webknjaz) [inline .github/workflows/ci.yaml:21]: I'm pretty sure they should be visible as a reference and fixed.

File: .github/workflows/ci.yaml
Comment (webknjaz) [inline .github/workflows/ci.yaml:27]: This should be one.

File: .github/workflows/ci.yaml
Comment (webknjaz) [inline .github/workflows/ci.yaml:4]: No need for annotations.

File: .github/workflows/ci.yaml
Comment (ssbarnea) [reply .github/workflows/ci.yaml:21]: I would not add until we get the checks green and fixing would be outside the scope of this change. If we include them and
ignore checks all PRs will report red, making much harder
to identify those that are ready for review or need more work.

That is due to how checks work, any red one makes the entire check red, even if is not required for merging. Its purely due
to the visual hint.

I added bug for tracking that #26

**Pull request closed without merging**
